contracts/feature-tests/composability/interact/src/composability_interact_state.rs

```diff
@@ -0,0 +1,91 @@
+use crate::{ContractInfo, DebugApi};
+use serde::{Deserialize, Serialize};
+use std::{
+    io::{Read, Write},
+    path::Path,
+};
+
+/// Default contract address
+const DEFAULT_CONTRACT_ADDRESS: &str =
+    "0x0000000000000000000000000000000000000000000000000000000000000000";
+
+/// State file
+const STATE_FILE: &str = "state.toml";
+
+pub type VaultContract = ContractInfo<vault::Proxy<DebugApi>>;
+pub type ForwarderRawContract = ContractInfo<forwarder_raw::Proxy<DebugApi>>;
+pub type PromisesContract = ContractInfo<promises_features::Proxy<DebugApi>>;
+
+/// Composability Interact state
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct State {
+    vault_address: Option<String>,
+    forwarder_raw_address: Option<String>,
+    promises_address: Option<String>,
+}
+
+impl State {
+    // Deserializes state from file
+    pub fn load_state() -> Self {
+        if Path::new(STATE_FILE).exists() {
+            let mut file = std::fs::File::open(STATE_FILE).unwrap();
+            let mut content = String::new();
+            file.read_to_string(&mut content).unwrap();
+            toml::from_str(&content).unwrap()
+        } else {
+            Self::default()
+        }
+    }
+
+    /// Sets the vault address
+    pub fn set_vault_address(&mut self, address: &str) {
+        self.vault_address = Some(String::from(address));
+    }
+    /// Sets the forwarder-raw address
+    pub fn set_forwarder_raw_address(&mut self, address: &str) {
+        self.forwarder_raw_address = Some(String::from(address));
+    }
+    /// Sets the promises address
+    pub fn set_promises_address(&mut self, address: &str) {
+        self.promises_address = Some(String::from(address));
+    }
+
+    /// Returns the vault contract
+    pub fn _vault(&self) -> VaultContract {
```

Comment:
no need to prefix method name with '_'

Reply:
I don't use this method and I used `_` to silence clippy. Will remove it when I call the method.

Reply:
You can also do `#[allow(dead_code)]`, gets rid of all warnings